### PR TITLE
Introduce Sketch block

### PR DESCRIPTION
**Pull request by oskosk**
Introduces a freehand sketching block.

Features:

* [x] Provide controls for three brush presets.
* [x] Provide controls with fixed colors from the theme palette.
* [x] Clear the whole sketch
* [x] Provide a resizer for block height
* [x] Provide alignment controls
* [x] Update description in readme.txt (currently showing Picasso's quote)
* [x] Update plugin metadata in header (currently showing Picasso's quote)
* [ ] Update images for icon
* [x] Update images for banner
* [x] Update screenshot images
####

Testing instructions

1. Clone this branch
2. Run `yarn build`
3. Run `yarn env start`
4. Visit the locally launched site, and log in
5. Create a new post and add a Sketch block

image

image

### CODE REVIEW

**Comment by jasmussen** [comment]
I see what you're doing here, and I love it 👌 

Let me know if/when you'd like a review and/or visual help. For example might I offer a better "clear" icon?

**Comment by oskosk** [comment]
> For example might I offer a better "clear" icon?

Definitely, although @pablohoneyhoney had another one to add here. They were proposed in a Figma doc but the clear one was a late addition.

**Comment by jasmussen** [comment]
If Pablo's got it, he's got it 👌

I'll just sit back and enjoy!

**Comment by pablohoneyhoney** [comment]
What's your suggestion @jasmussen? Maybe it's better 🙂

The trash can was a literal link to paper.

**Comment by jasmussen** [comment]
Honestly the trash can can be fine, in my rendering of the branch it was just a tad thin, it could just be collapsed vectors:

Screenshot 2021-05-04 at 15 13 39

I was thinking another literal metaphor, backspace:

Screenshot 2021-05-04 at 15 13 02

Screenshot 2021-05-04 at 15 12 56

Though I'm not sure how well that translates across writing directions 🤔

**Comment by pablohoneyhoney** [comment]
That trash icon isn't correct actually, per the new one in Figma. 

image

**Comment by jasmussen** [comment]
Cool, looks like the version that shipped in trunk was wrong, and I've updated that here: https://github.com/WordPress/gutenberg/pull/31463

so as soon as the icon component gets updated, it looks like it will be refreshed here as well!

**Comment by oskosk** [comment]
Thanks @jasmussen !

**Comment by jasmussen** [comment]
I may be compiling this branch wrong, but I'm seeing a crash on the color palettes. I'm also seeing an outlined trash icon:

Screenshot 2021-06-15 at 10 41 46

Still very excited about this block!

**Comment by mtias** [comment]
Alright, this is in a better place now

image

File: blocks/sketch/index.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+define( __NAMESPACE__ . '\DEFAULT_HEIGHT', 450);
+
+
+function a8c_sketch_render( $attributes ) {
+	$strokes = $attributes['strokes'] ?? [];
+	$align   = $attributes['align'] ?? '';
+	$height   = $attributes['height'] ?? DEFAULT_HEIGHT;
+	// The class name that affects alignment is called alignwide, alignfull, etc
+	$align = $align ? " align$align" : '';
```
Comment (mtias) [inline blocks/sketch/index.php:12]: Using the "supports" don't you get all this for free?

File: blocks/sketch/src/save.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+import { __ } from '@wordpress/i18n';
+import { useBlockProps } from '@wordpress/block-editor';
+
+const Save = () => {
+	const blockProps = useBlockProps.save();
+
+	return (
+		<figure { ...blockProps }>
+			{ __( 'Sketch drawing', 'sketch' ) }
```
Comment (mtias) [inline blocks/sketch/src/save.js:16]: What should we be saving here?

File: bundler/bundles/sketch.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+	"blocks": [ "sketch" ],
+	"version": "1.0.0",
+	"name": "Sketch Block",
+	"description": "“There's nothing more difficult than a line.” — Pablo Picasso",
```
Comment (mtias) [inline bundler/bundles/sketch.json:5]: need to update this one

File: package.json [hunk 4]
```diff
@@ -58,7 +58,8 @@
 		"@wordpress/block-editor": "^5.3.0",
 		"@wordpress/blocks": "^8.0.0",
 		"classnames": "^2.3.1",
-		"fast-average-color": "^6.4.0",
+		"fast-average-color": "^6.3.0",
```
Comment (mtias) [inline package.json:61]: bump?

Comment (oskosk) [reply package.json:61]: There's a rebase needed. Was about to do it

File: bundler/bundles/sketch.json (hunk 3 above)
Comment (oskosk) [reply bundler/bundles/sketch.json:5]: Nothing for now. Addressed in c47ddd8ddce06a76eef4d381944b144f946851b1

File: blocks/sketch/src/save.js (hunk 2 above)
Comment (oskosk) [reply blocks/sketch/src/save.js]: Nothing for now. Addressed in c47ddd8

File: blocks/sketch/index.php (hunk 1 above)
Comment (oskosk) [reply blocks/sketch/index.php:12]: We only get those for free when rendering the same markup saved by the `save()` method. Given that we're not going to save any significan markup for now (c47ddd8) we're recreating the markup with classes on the PHP side

**Comment by oskosk** [comment]
Merging. The plugin has been submitted for review 🎉

**Pull request merged**